control/metrics.go

```diff
@@ -357,7 +357,25 @@ func validateMetricNamespace(ns core.Namespace) error {
 	return nil
 }
 
+// validateDynamic validates a dynamic namespace qualification.
+// A dynamic element requires the name field while a static element does not.
+func validateDynamic(ns core.Namespace) error {
```

Comment:
I find the function name a bit confusing. To me it suggests that the function validates that the namespace is dynamic. Couldn't this validation be part of the `validateMetricNamespace` function instead?